src/sleep/sleep.c

```diff
@@ -263,41 +263,108 @@ static int execute(
 }
 
 static int execute_s2h(const SleepConfig *sleep_config) {
-        _cleanup_close_ int tfd = -1;
-        struct itimerspec ts = {};
+        usec_t suspend_interval = sleep_config->hibernate_delay_sec, before_timestamp, after_timestamp;
         int r;
 
         assert(sleep_config);
 
-        tfd = timerfd_create(CLOCK_BOOTTIME_ALARM, TFD_NONBLOCK|TFD_CLOEXEC);
-        if (tfd < 0)
-                return log_error_errno(errno, "Error creating timerfd: %m");
+        do {
+                _cleanup_close_ int tfd = -1;
+                struct itimerspec ts = {};
+                int last_capacity, current_capacity, previous_discharge_rate, estimated_discharge_rate = 0;
+                double time_elapsed;
+
+                tfd = timerfd_create(CLOCK_BOOTTIME_ALARM, TFD_NONBLOCK|TFD_CLOEXEC);
+                if (tfd < 0)
+                        return log_error_errno(errno, "Error creating timerfd: %m");
+
+                /* Store current battery capacity and current time before suspension*/
+                r = read_battery_capacity_percentage();
+                if (r > 0) {
+                        last_capacity = r;
+                        before_timestamp = now(CLOCK_BOOTTIME);
+                }
+                else if (r == -ENOENT)
+                        log_debug_errno(r, "Battery capacity percentage unavailable, System will be suspended for hibernatedelaysec interval: %m");
```

Comment:
maybe print the actual value rather than `hibernatedelaysec`

Reply:
yeah makes sense